### PR TITLE
pv: add the $env() PV class to read env vars

### PR DESCRIPTION
**Pull request by camilleoudot**
this allows kamailio to have read-only (for now) access to environment variables

### CODE REVIEW

File: src/modules/pv/pv_core.c [hunk 1]
```diff
@@ -3233,3 +3235,44 @@ int pv_get_msg_attrs(sip_msg_t *msg, pv_param_t *param, pv_value_t *res)
...
+{
+	char *val;
+	char *csname = (char *) param->pvn.u.dname;
+
+	if (csname) {
+		val = getenv(csname);
+
+		if (val) {
+			return pv_get_strzval(msg, param, res, val);
+		} else {
```
Comment (linuxmaniac) [inline src/modules/pv/pv_core.c]: what about

```
if (csname) {
  val = getenv(csname);
  if(val) return pv_get_strzval(msg, param, res, val);
}
return pv_get_null(msg, param, res);
```

File: src/modules/pv/pv_core.c [hunk 2]
```diff
@@ -3233,3 +3235,44 @@ int pv_get_msg_attrs(sip_msg_t *msg, pv_param_t *param, pv_value_t *res)
 			return pv_get_null(msg, param, res);
 	}
 }
+
+int pv_parse_env_name(pv_spec_p sp, str *in)
+{
+	if(in->s==NULL || in->len<=0)
+		return -1;
+
+	char *csname;
```
Comment (linuxmaniac) [inline src/modules/pv/pv_core.c]: can you please move the declaration to the top, please.

File: src/modules/pv/pv_core.c (hunk 1 above)
Comment (camilleoudot) [inline src/modules/pv/pv_core.c]: easier to read, thanks

File: src/modules/pv/pv_core.c (hunk 2 above)
Comment (camilleoudot) [inline src/modules/pv/pv_core.c]: sure

File: src/modules/pv/pv_core.c [hunk 3]
```diff
@@ -3238,11 +3238,11 @@ int pv_get_msg_attrs(sip_msg_t *msg, pv_param_t *param, pv_value_t *res)
 
 int pv_parse_env_name(pv_spec_p sp, str *in)
 {
+	char *csname;
+
 	if(in->s==NULL || in->len<=0)
```
Comment (linuxmaniac) [inline src/modules/pv/pv_core.c]: I would do
```
if(sp==NULL || in == NULL || in->len <=0)
``` 

as in the [kamailio-devel-guide](http://www.asipto.com/pub/kamailio-devel-guide/#c16_add_pv)

Comment (camilleoudot) [inline src/modules/pv/pv_core.c]: it does not look like any of the parameters can be null, judging by the checks done before calling this function in https://github.com/kamailio/kamailio/blob/master/src/core/pvapi.c#L873, but I may be missing something else

Comment (linuxmaniac) [inline src/modules/pv/pv_core.c]: and where is the documentation update?

Comment (camilleoudot) [inline src/modules/pv/pv_core.c]: I'll document it on the wiki when/if it makes its way to master eventually (after the feature freeze?)

Comment (linuxmaniac) [inline src/modules/pv/pv_core.c]: FTR, it works nicely 
```
route[TEST] {
	$var(tmp) = $env(FAKE);
	xlog("L_NOTICE", "PATH=$env(PATH) FAKE=$var(tmp)\n");
	return;
}
```
```
 3(1978) NOTICE: <script>: PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin FAKE=0
```

**Comment by miconda** [comment]
Thanks! Can be merged into master.

**Pull request merged**